sbnana/SBNAna/NuMINumuXSec/ICARUSNumuXsec_Variables.h

```diff
@@ -0,0 +1,1315 @@
+
+    //cout << "[varGetMatchedTrackForProton] Truth proton genE = " << slc->truth.prim.at(truth_index).genE << endl;
+    double truth_E = slc->truth.prim.at(truth_index).genE;
+
+    int PTrackInd(-1);
+    for(std::size_t i(0); i < slc->reco.trk.size(); ++i){
+      auto const& trk = slc->reco.trk.at(i);
+      if(isnan(trk.truth.p.genE)) continue;
+      double this_matched_E = trk.truth.p.genE;
+      if( abs(trk.truth.p.pdg)==13 && fabs(truth_E-this_matched_E)/truth_E < 0.01 ){
+        PTrackInd = i;
+      }
+    }
+    if(PTrackInd<0) return -999.;
+    //cout << "[varGetMatchedTrackForProton] Truth proton genE = " << slc->truth.prim.at(truth_index).genE << endl;
+    //cout << "[varGetMatchedTrackForProton] Found track's matched genE = " << slc->reco.trk.at(PTrackInd).truth.p.genE << endl;
+
+    return PTrackInd;
+
+  });
+
+  const Var varTruthMuonMatchedTrackContainedness([](const caf::SRSliceProxy* slc) -> int {
+    int muonTrackIndex = varTruthMuonMatchedTrackIndex(slc);
+    //==== -1 : No track found
+    //==== 0 : Exiting
+    //==== 1 : Contained
+    if(muonTrackIndex>=0){
+
+      auto const& trk_Muon = slc->reco.trk.at(muonTrackIndex);
+
+      double XMargin = 25.;
+      double YMargin = 25.;
+      double ZMarginUp = 30.;
+      double ZMarginDown = 50.;
+
+      bool isContained;
+      //==== Cryo0
+      if( trk_Muon.end.x < 0 ){
+        isContained = ( !isnan(trk_Muon.end.x) &&
+                      ( trk_Muon.end.x < -71.1 - XMargin && trk_Muon.end.x > -369.33 + XMargin ) &&
+                      !isnan(trk_Muon.end.y) &&
+                      ( trk_Muon.end.y > -181.7 + YMargin && trk_Muon.end.y < 134.8 - YMargin ) &&
+                      !isnan(trk_Muon.end.z) &&
+                      ( trk_Muon.end.z > -895.95 + ZMarginUp && trk_Muon.end.z < 895.95 - ZMarginDown ) );
+      }
+      //==== Cryo1
+      else{
+        isContained = ( !isnan(trk_Muon.end.x) &&
+                      ( trk_Muon.end.x > 71.1 + XMargin && trk_Muon.end.x < 369.33 - XMargin ) &&
+                      !isnan(trk_Muon.end.y) &&
+                      ( trk_Muon.end.y > -181.7 + YMargin && trk_Muon.end.y < 134.8 - YMargin ) &&
+                      !isnan(trk_Muon.end.z) &&
+                      ( trk_Muon.end.z > -895.95 + ZMarginUp && trk_Muon.end.z < 895.95 - ZMarginDown ) );
+      }
+      if(isContained) return 1;
+      else return 0;
+    }
+    else{
+      return -1;
+    }
+
+  });
+
+  const Var varTruthMuonMatchedTrackChi2Proton([](const caf::SRSliceProxy* slc) -> double {
+    int muonTrackIndex = varTruthMuonMatchedTrackIndex(slc);
+    if(muonTrackIndex>=0){
+
+      auto const& trk = slc->reco.trk.at(muonTrackIndex);
+      float Chi2Proton(-999.);
+      if(trk.bestplane == 0){
+        Chi2Proton = trk.chi2pid0.chi2_proton;
+      }
+      else if(trk.bestplane == 1){
+        Chi2Proton = trk.chi2pid1.chi2_proton;
+      }
+      else{
+        Chi2Proton = trk.chi2pid2.chi2_proton;
+      }
```

Comment:
All of this ugliness will eventually be fixed by https://github.com/SBNSoftware/sbnana/pull/41 which makes these itno proper arrays.